currencies/rate_converter.go

```diff
@@ -168,12 +170,24 @@ func (rc *RateConverter) LastUpdated() time.Time {
 
 // Rates returns current conversions rates
 func (rc *RateConverter) Rates() Conversions {
-	if rates := rc.rates.Load(); rates != (*Rates)(nil) && rates != nil {
+	if rates := rc.rates.Load(); rc.CheckRatesCleared(rates) && rates != nil {
```

Comment:
I didn't want to pass `rates` in here but I also didn't want to call `rc.rates.Load()` a second time in this method because it seems like unnecessary work. Given that fact, I'm not sure how much added value we get here by the self documenting method name (which I was initially on board with when we discussed) since we have introduced a little awkwardness into this method call.

Reply:
Two options here. You can write the helper method a bit differently, as you don't need to add a RateConverter receiver. ie:
```
func checkRatesCleared(rates interface{}) bool {
	// atomic.Value field rates is an empty interface and will be of type *Rates the first time rates are stored
	return rates != (*Rates)(nil)
}
```

.. or remove the helper method if you feel it hurts the flow. Whichever you think is best.